internal/controllers/import_controller_v3_test.go

```diff
@@ -0,0 +1,441 @@
+		Expect(err).ToNot(HaveOccurred())
+		ns.Labels = map[string]string{
+			importLabelName: "true",
+		}
+		Expect(cl.Update(ctx, ns)).To(Succeed())
+
+		sampleTemplate = setTemplateParams(
+			testdata.ImportManifest,
+			map[string]string{"${TEST_CASE_NAME}": "mgmtv3"},
+		)
+
+		r = &CAPIImportManagementV3Reconciler{
+			Client:             cl,
+			RancherClient:      cl, // rancher and rancher-turtles deployed in the same cluster
+			remoteClientGetter: remote.NewClusterClient,
+			Scheme:             testEnv.GetScheme(),
+		}
+
+		capiCluster = &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      capiClusterName,
+				Namespace: ns.Name,
+			},
+		}
+
+		rancherCluster = &managementv3.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:    capiCluster.Namespace,
+				GenerateName: "c-",
+				Labels: map[string]string{
+					capiClusterOwner:          capiCluster.Name,
+					capiClusterOwnerNamespace: capiCluster.Namespace,
+					ownedLabelName:            "",
+				},
+			},
+		}
+
+		rancherClusters = &managementv3.ClusterList{}
+
+		selectors = []client.ListOption{
+			client.MatchingLabels{
+				capiClusterOwner:          capiCluster.Name,
+				capiClusterOwnerNamespace: capiCluster.Namespace,
+			},
+		}
+
+		clusterRegistrationToken = &managementv3.ClusterRegistrationToken{
+			ObjectMeta: metav1.ObjectMeta{},
+		}
+
+		capiKubeconfigSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-kubeconfig", capiCluster.Name),
+				Namespace: ns.Name,
+			},
+			Data: map[string][]byte{
+				secret.KubeconfigDataName: kubeConfigBytes,
+			},
+		}
+	})
+
+	AfterEach(func() {
+		objs, err := manifestToObjects(strings.NewReader(sampleTemplate))
+		clientObjs := []client.Object{
+			capiCluster,
+			rancherCluster,
+			clusterRegistrationToken,
+			capiKubeconfigSecret,
+		}
+		for _, obj := range objs {
+			clientObj, ok := obj.(client.Object)
+			Expect(ok).To(BeTrue())
+			clientObjs = append(clientObjs, clientObj)
+		}
+		Expect(err).ToNot(HaveOccurred())
+		Expect(test.CleanupAndWait(ctx, cl, clientObjs...)).To(Succeed())
+		Expect(testEnv.Cleanup(ctx, ns)).To(Succeed())
+		for _, cluster := range rancherClusters.Items {
+			testEnv.Cleanup(ctx, &cluster)
+			testEnv.Cleanup(ctx, &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{Name: cluster.Name},
+			},
+			)
+		}
+	})
+
+	It("should reconcile a CAPI cluster when control plane not ready", func() {
+		Expect(cl.Create(ctx, capiCluster)).To(Succeed())
+
+		Eventually(ctx, func(g Gomega) {
+			res, err := r.Reconcile(ctx, reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: capiCluster.Namespace,
+					Name:      capiCluster.Name,
+				},
+			})
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(res.RequeueAfter).To(Equal(defaultRequeueDuration))
+		}).Should(Succeed())
+	})
+
+	It("should reconcile a CAPI cluster when rancher cluster doesn't exist", func() {
+		capiCluster.Labels = map[string]string{
```

Comment:
I think here a namespace import label should be removed, so that this statement would be tested.